nsdb-cluster/src/main/scala/io/radicalbit/nsdb/cluster/logic/LocalityReadNodesSelection.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018-2020 Radicalbit S.r.l.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.radicalbit.nsdb.cluster.logic
+
+import io.radicalbit.nsdb.model.Location
+
+/**
+  * ReadNodesSelection strategy that privileges local locations.
+  * @param localNode the local node.
+  */
+class LocalityReadNodesSelection(localNode: String) extends ReadNodesSelection {
+
+  override def getDistinctLocationsByNode(locationsWithReplicas: Seq[Location]): Map[String, Seq[Location]] = {
+    locationsWithReplicas
+      .groupBy(l => (l.from, l.to))
+      .collect {
```

Comment:
this may be a map?

Reply:
yes. Good point. Since there is a total function involved it must be a map ( I've started the implementation thinking about a partial function). I'll fi it in the next PR. Thanks